W-21747880: Create Otel Metrics public documentation. (#3326)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/otel-support.adoc

| Counter
| Number of sub-flows and private flows (no message source)

| `active-private-flows`
| Counter
| Private flows currently in a started state

| `declared-apikit-flows`
| Counter
| Number of flows generated by APIKit from an API specification

| `active-apikit-flows`
| Counter
| APIKit flows currently in a started state
|===

Each metric also has a `-v2` variant (e.g., `active-trigger-flows-v2`) that currently reports the same values. The difference between declared and active counts can reveal flows that failed to start or were intentionally stopped.

=== Error Classification Metrics

_Instrumentation scope: `Mule runtime error metrics`_

These metrics break down runtime errors by their Mule error type, providing more granular error visibility than the aggregate `execution.errors` counter.

[cols="1,1,1,3", options="header"]
|===
| Metric | Type | Attributes | Description

| `error-count`
| Counter
| `error-type`, `mule.artifact.id`
| Count of errors classified by Mule error type
|===

The `error-type` attribute contains the Mule error identifier (e.g., `MULE:COMPOSITE_ROUTING`, `MULE:UNKNOWN`, `MULE:SOURCE_RESPONSE_SEND`), enabling per-type trend analysis and alerting.

> NOTE: `error-count` and `execution.errors` are tracked independently. `execution.errors` may report a higher total because it counts all execution-level errors including those not classified into a specific error type.

Reviewer: ```suggestion
NOTE: `error-count` and `execution.errors` are tracked independently. `execution.errors` can report a higher total because it counts all execution-level errors, including those not classified into a specific error type.
```

|===
| Metric | Type | Description

| `jvm.memory.used`
| Gauge
| Current memory usage by pool (heap/non-heap)

| `jvm.memory.committed`
| Gauge
| Memory committed by the JVM per pool

| `jvm.memory.limit`
| Gauge
| Maximum memory available per pool

| `jvm.memory.used_after_last_gc`
| Gauge
| Memory used after the most recent GC

| `jvm.gc.duration`
| Histogram
| Garbage collection pause durations by GC name

| `jvm.thread.count`
| Gauge
| Active threads by state

| `jvm.class.count`
| Gauge
| Currently loaded classes

| `jvm.class.loaded`
| Counter
| Total classes loaded since JVM start

| `jvm.class.unloaded`
| Counter
| Total classes unloaded since JVM start

| `jvm.cpu.count`
| Gauge
| Available processors

| `jvm.cpu.recent_utilization`
| Gauge
| Recent CPU utilization (0.0 to 1.0)

| `jvm.cpu.time`
| Counter
| Cumulative CPU time consumed
|===

=== Direct Telemetry Stream Exporter Metrics

_Instrumentation scopes: `io.opentelemetry.exporters.otlp_grpc_span_exporter`, `io.opentelemetry.exporters.otlp_grpc_log_exporter`, `io.opentelemetry.exporters.otlp_grpc_metric_exporter`, `io.opentelemetry.sdk.metrics`, `io.opentelemetry.sdk.trace`, `io.opentelemetry.sdk.logs`_

> These metrics provide visibility into the health and throughput of the Direct Telemetry Stream exporters, allowing you to monitor whether telemetry data is being successfully exported.

Reviewer: ```suggestion
These metrics show the health and throughput of the Direct Telemetry Stream exporters so you can monitor whether they successfully export telemetry data.
```

| `mule.openTelemetry.meter.exporter.backoff.initial` | 1 second | Wait time before the first retry attempt.
| `mule.openTelemetry.meter.exporter.backoff.maximum` | 5 seconds | Maximum wait time before a retry attempt.
| `mule.openTelemetry.meter.exporter.backoff.multiplier` | 1.5 | Multiplier applied to the last wait time before a new retry attempt.
|===

=== Metrics Temporality

Mule runtime exports all its metrics using cumulative https://opentelemetry.io/docs/specs/otel/metrics/data-model/#temporality[temporality]. Each exported data point represents the total accumulated value of a metric since the application started, not just the delta since the last export.

If an export fails (including after backpressure retries), that export cycle is dropped without being queued or persisted. Because metrics are cumulative, a dropped export doesn't result in permanent data loss. The next successful export carries the full accumulated state, and the backend service reconstructs the timeline with only a brief gap in resolution.

> If the Direct Telemetry Stream exporter is still engaged in a previous metric export cycle when the next interval fires, the new cycle is skipped and a `"Exporter busy. Dropping metrics."` `INFO` log message is triggered. To minimize skipped cycles, ensure your collector and backend service are reachable with low latency.

Reviewer: ```suggestion
If the Direct Telemetry Stream exporter is still engaged in a previous metric export cycle when the next interval fires, the new cycle is skipped and a `"Exporter busy. Dropping metrics."` `INFO` log message is triggered. To minimize skipped cycles, keep your collector and backend service reachable with low latency.
```
